jar-infer/jar-infer-lib/src/test/java/com/uber/nullaway/jarinfer/EntriesComparator.java

```diff
@@ -0,0 +1,107 @@
+   *
+   * @param jarFile1 Path to the first jar file.
+   * @param jarFile2 Path to the second jar file.
+   * @return True iff the entries present in the two jar files are the same.
+   * @throws IOException
+   */
+  public static boolean compareEntriesInJars(String jarFile1, String jarFile2) throws IOException {
+    Preconditions.checkArgument(jarFile1.endsWith(".jar"), "invalid jar file: " + jarFile1);
+    Preconditions.checkArgument(jarFile2.endsWith(".jar"), "invalid jar file: " + jarFile2);
+    JarFile jar1 = new JarFile(jarFile1);
+    Set<String> jar1Entries = new HashSet<>();
+    for (Enumeration<JarEntry> entries = jar1.entries(); entries.hasMoreElements(); ) {
+      jar1Entries.add(entries.nextElement().getName());
+    }
+    JarFile jar2 = new JarFile(jarFile2);
+    Set<String> jar2Entries = new HashSet<>();
+    for (Enumeration<JarEntry> entries = jar2.entries(); entries.hasMoreElements(); ) {
+      jar2Entries.add(entries.nextElement().getName());
+    }
+    return jar1Entries.equals(jar2Entries);
+  }
+
+  /**
+   * Compares the entries in the given 2 aar files and the entries in "classes.jar" in them.
+   * However, this function does not compare the contents of the entries themselves.
+   *
+   * @param aarFile1 Path to the first aar file.
+   * @param aarFile2 Path to the second aar file.
+   * @return True iff the entries present in the two aar files are the same and entries in
+   *     "classes.jar" in the two aar files are the same.
+   * @throws IOException
+   */
+  public static boolean compareEntriesInAars(String aarFile1, String aarFile2) throws IOException {
+    Preconditions.checkArgument(aarFile1.endsWith(".aar"), "invalid aar file: " + aarFile1);
+    Preconditions.checkArgument(aarFile2.endsWith(".aar"), "invalid aar file: " + aarFile2);
+    ZipFile zip1 = new ZipFile(aarFile1);
+    Set<String> zip1Entries = new HashSet<>();
+    for (Enumeration<? extends ZipEntry> entries = zip1.entries(); entries.hasMoreElements(); ) {
+      zip1Entries.add(entries.nextElement().getName());
+    }
+    ZipFile zip2 = new ZipFile(aarFile2);
+    Set<String> zip2Entries = new HashSet<>();
+    for (Enumeration<? extends ZipEntry> entries = zip2.entries(); entries.hasMoreElements(); ) {
+      zip2Entries.add(entries.nextElement().getName());
+    }
+    if (!zip1Entries.equals(zip2Entries)) {
+      return false;
+    }
+
+    // Check if all the entries in "classes.jar" in both aar files are the same.
+    // We expect to find a classes.jar entry in the aar files.
+    ZipEntry zip1Jar = zip1.getEntry(classesJarInAar);
+    ZipEntry zip2Jar = zip2.getEntry(classesJarInAar);
+    if (zip1Jar == null || zip2Jar == null) {
+      return false;
+    }
+    JarInputStream jarIS1 = new JarInputStream(zip1.getInputStream(zip1Jar));
+    JarInputStream jarIS2 = new JarInputStream(zip2.getInputStream(zip2Jar));
+    Set<String> jar1Entries = new HashSet<>();
+    JarEntry jar1Entry = jarIS1.getNextJarEntry();
```

Comment:
Also looks like sort of duplicated code to me, but in favor of it because this is in test code and good for test case readability.

Reply:
Same as above. Leaving it as is for now.